lightning/src/ln/channelmanager.rs

```diff
@@ -4259,9 +4231,9 @@ where
 			expected_amt_msat = htlc.total_value_received;
 
 			if let OnionPayload::Spontaneous(_) = &htlc.onion_payload {
-				// We don't currently support MPP for spontaneous payments, so just check
+				// If the user chooses not to support MPP for spontaneous payments, just check
 				// that there's one payment here and move on.
-				if sources.len() != 1 {
+				if !self.default_configuration.accept_mpp_keysend && sources.len() != 1 {
 					log_error!(self.logger, "Somehow ended up with an MPP spontaneous payment - this should not be reachable!");
 					debug_assert!(false);
 					valid_mpp = false;
```

Comment:
Oh true! Also I realized it's actually reachable just generally if our config rejects mpp keysend since the config only means we'll reject them here while claiming, so yea logging an error doesn't really make sense